### ISSUE TITLE
Error when importing a file with filename matching import*.sol

### ISSUE THREAD

**Issue by adlerjohn** [trufflesuite/truffle#3348]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

An error is raised when importing a Solidity file with filename matching `import*.sol` (i.e. lowercase "import" followed by anything).

## Steps to Reproduce

A repository to reproduce the bug can be found here: https://github.com/adlerjohn/truffle-import-bug-1. Alternatively, the contracts below will reproduce the bug.

`TopLevel.sol`:
```solidity
import "./import.sol";
contract TopLevel {}
```

`import.sol`:
```solidity
contract Import {}
```

## Expected Behavior

Contract compiles properly with no errors.

## Actual Results

```sh
Compiling your contracts...
===========================
Error: Could not find  not found: File import callback not supported
import  from any sources; imported from /home/truffle-import-bug-1/contracts/TopLevel.sol
    at Resolver.<anonymous> (/home/truffle-import-bug-1/node_modules/truffle/build/webpack:/packages/resolver/dist/lib/resolver.js:54:1)
    at Generator.next (<anonymous>)
    at fulfilled (/home/truffle-import-bug-1/node_modules/truffle/build/webpack:/packages/resolver/dist/lib/resolver.js:5:42)
Truffle v5.1.43 (core: 5.1.43)
Node v12.18.3
```

## Environment

* Operating System: Ubuntu 20.04
* Ethereum client: Whatever truffle has builtin
* Truffle version (`truffle version`): `v5.1.43 (core: 5.1.43)`
* Solidity version: `0.6.12 (solc-js)`
* node version (`node --version`): `v12.18.3`
* npm version (`npm --version`): `6.14.6`

**Comment by eggplantzzz** [comment]
Ah very weird, thanks for the report. We'll have to look into it and see what's going on here!

**Comment by gpersoon** [comment]
I also have the same issue (on windows 10).
I guess it is related to this line:
https://github.com/trufflesuite/truffle/blob/develop/packages/compile-solidity/parser.js#L67

Where the keyword "import" has a special meaning.

**Comment by adlerjohn** [comment]
Closing to clean up my issue list. No one uses truffle anymore anyways, so this is no longer relevant.
